CMakeLists.txt

```diff
@@ -0,0 +1,934 @@
+#
+#	Copyright (c) 2020 Sibi Siddharthan
+#
+cmake_minimum_required(VERSION 3.15)
+project(git 
+		VERSION 2.26
+		LANGUAGES C)
+
+#[[
+NOTE:
+FOR LINUX USE THIS GIT REPO https://github.com/git/git
+FOR WINDOWS USE THIS GIT REPO https://github.com/git-for-windows/git
+IF YOU MIX UP REPOS THINGS MIGHT NOT WORK AS EXPECTED
+TRY YOUR LUCK
+]]
+
+#TODO gitk git-gui gitweb
+#TODO Enable NLS on windows natively
+#TODO Add pcre support
+#TODO exclude tests which are not supported
+
+include(CheckTypeSize)
+include(CheckCSourceRuns)
+include(CheckCSourceCompiles)
+include(CheckIncludeFile)
+include(CheckFunctionExists)
+include(CheckSymbolExists)
+include(CheckStructHasMember)
+include(CTest)
+
+find_package(ZLIB REQUIRED)
+find_package(CURL)
+find_package(EXPAT)
+find_package(Iconv)
+
+if(NOT WIN32)
+find_package(Intl)
+endif()
+
+if(NOT Intl_FOUND)
+add_compile_definitions(NO_GETTEXT)
+if(NOT Iconv_FOUND)
+add_compile_definitions(NO_ICONV)
+endif()
+endif()
+
+if(WIN32)
+add_compile_definitions(NO_GETTEXT)
+endif()
+
+include_directories(SYSTEM ${ZLIB_INCLUDE_DIRS})
+if(CURL_FOUND)
+	include_directories(SYSTEM ${CURL_INCLUDE_DIRS})
+endif()
+if(EXPAT_FOUND)
+	include_directories(SYSTEM ${EXPAT_INCLUDE_DIRS})
+endif()
+if(Iconv_FOUND)
+	include_directories(SYSTEM ${Iconv_INCLUDE_DIRS})
+endif() 
+if(Intl_FOUND)
+	include_directories(SYSTEM ${Intl_INCLUDE_DIRS})
+endif() 
+
+find_program(SH_EXE sh)
+if(NOT SH_EXE)
+	message(FATAL_ERROR "sh interpreter was not found in your path,please install one.On Windows you can get it from here https://gitforwindows.org/")
+endif()
+
+if(WIN32)
+	find_program(WINDRES_EXE windres)
+	if(NOT WINDRES_EXE)
+		message(FATAL_ERROR "Install windres on Windows for resource files")
+	endif()
+endif()
+
+find_program(MSGFMT_EXE msgfmt)
+if(NOT MSGFMT_EXE)
+	message(WARNING "Text Translations won't be build")
+endif()
+
+
+#Force all visual studio outputs to CMAKE_BINARY_DIR
+if(CMAKE_C_COMPILER_ID STREQUAL "MSVC")
+	set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG ${CMAKE_BINARY_DIR})
+	set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE ${CMAKE_BINARY_DIR})
+	add_compile_options(/MP)
+endif()
+
+#note SHA1_DC SHA1_BLK mutually exclusive
+#default behaviour
+add_compile_definitions(GIT_HOST_CPU="x86_64")
```

Comment:
No there isn't one. You have to use uname for this. This assumes you have a working uname in the git-sdk-minimal which is in the path on windows.
CMAKE_SYSTEM_PROCESSOR gives different results depending on the OS
eg. Ubuntu -> x86_64, Windows -> AMD64
To do this we need to see what the variable returns for each OS, for each preocessor. This might also be cmake version dependent(Not sure but can be the case).
One way is to define the architecture in the command line.

Reply:
As the information is used purely for informational purposes (`git version --build-options`), I would think that showing `x86_64` on some, and `AMD64` on others, is okay.

It's not okay, of course, to show either when cross-compiling, say, for ARM64.